pkg/sdk/schemas.go

```diff
@@ -13,3 +50,324 @@ func (v *Schema) ID() SchemaIdentifier {
+	schema               bool             `ddl:"static" sql:"SCHEMA"` //lint:ignore U1000 This is used in the ddl tag
+	IfExists             *bool            `ddl:"keyword" sql:"IF EXISTS"`
+	name                 SchemaIdentifier `ddl:"identifier"`
+	NewName              SchemaIdentifier `ddl:"identifier" sql:"RENAME TO"`
+	SwapWith             SchemaIdentifier `ddl:"identifier" sql:"SWAP WITH"`
+	Set                  *SchemaSet       `ddl:"list,no_parentheses" sql:"SET"`
+	Unset                *SchemaUnset     `ddl:"list,no_parentheses" sql:"UNSET"`
+	EnableManagedAccess  *bool            `ddl:"keyword" sql:"ENABLE MANAGED ACCESS"`
+	DisableMangaedAccess *bool            `ddl:"keyword" sql:"DISABLE MANAGED ACCESS"`
+}
+
+func (opts *AlterSchemaOptions) validate() error {
+	var errs []error
+	if !validObjectidentifier(opts.name) {
+		errs = append(errs, ErrInvalidObjectIdentifier)
+	}
+	if !exactlyOneValueSet(opts.NewName, opts.SwapWith, opts.Set, opts.Unset, opts.EnableManagedAccess, opts.DisableMangaedAccess) {
+		errs = append(errs, errors.New("Only one of the fields [ NewName | SwapWith | Set | Unset | EnableManagedAcccess | DisableManagedAcccess ] can be set at once"))
+	}
+	if valueSet(opts.Set) {
+		if err := opts.Set.validate(); err != nil {
+			errs = append(errs, err)
+		}
+	}
+	if valueSet(opts.Unset) {
+		if err := opts.Unset.validate(); err != nil {
+			errs = append(errs, err)
+		}
+	}
+	return errors.Join(errs...)
+}
+
+type SchemaSet struct {
+	DataRetentionTimeInDays    *int             `ddl:"parameter" sql:"DATA_RETENTION_TIME_IN_DAYS"`
+	MaxDataExtensionTimeInDays *int             `ddl:"parameter" sql:"MAX_DATA_EXTENSION_TIME_IN_DAYS"`
+	DefaultDDLCollation        *string          `ddl:"parameter,single_quotes" sql:"DEFAULT_DDL_COLLATION"`
+	Comment                    *string          `ddl:"parameter,single_quotes" sql:"COMMENT"`
+	Tag                        []TagAssociation `ddl:"keyword" sql:"TAG"`
+}
+
+func (v *SchemaSet) validate() error {
+	if valueSet(v.Tag) && anyValueSet(v.DataRetentionTimeInDays, v.MaxDataExtensionTimeInDays, v.DefaultDDLCollation, v.Comment) {
+		return errors.New("Tag cannot be set with other options")
+	}
+	return nil
+}
+
+type SchemaUnset struct {
+	DataRetentionTimeInDays    *bool              `ddl:"keyword" sql:"DATA_RETENTION_TIME_IN_DAYS"`
+	MaxDataExtensionTimeInDays *bool              `ddl:"keyword" sql:"MAX_DATA_EXTENSION_TIME_IN_DAYS"`
+	DefaultDDLCollation        *bool              `ddl:"keyword" sql:"DEFAULT_DDL_COLLATION"`
+	Comment                    *bool              `ddl:"keyword" sql:"COMMENT"`
+	Tag                        []ObjectIdentifier `ddl:"keyword" sql:"TAG"`
+}
+
+func (v *SchemaUnset) validate() error {
+	if valueSet(v.Tag) && anyValueSet(v.DataRetentionTimeInDays, v.MaxDataExtensionTimeInDays, v.DefaultDDLCollation, v.Comment) {
+		return errors.New("Tag cannot be set with other options")
```

Comment:
cannot be unset?

Reply:
I meant tag as a field, maybe I'll change it to "Tag field cannot be..."